### PR TITLE
Exported descriptions from the Debian udd database

### PR DESCRIPTION
**Pull request by smoe**
This pull request is the fruit of a collaboration of @tillea, @matuskalas, @hmenager and mine at the current and past Debian Med Sprints. The tool edam.sh spawns a python query that is written into a json file.
The tool edamJson2biotools.py transforms that file in analogy to the
export that @matuskalas  and @bgruening  agreed upon for conda.
Both tools are located in https://github.com/bio-tools/biotoolsConnect

**Note:** Included are tools from Debian that have _source package name == binary package name_ and biotoolsID was present in both Debian and https://github.com/bio-tools/content at the time (2020-02-08/09)

### CODE REVIEW

**Comment by matuskalas** [comment]
Suggestions for a little bit of perfecting of the already beautiful YAML for the EDAM annotation, after discussing it F2F with @smoe. Applies also to https://wiki.debian.org/UpstreamMetadata#Edam_files. Things to take into account are:
* `version` of EDAM must be present, because the stable numeric IDs are not present
* `topics` under `edam:`
* plural in the attribute name, where a list expected
* a list of values is possible for:
  * `topics`
  * `scope`-s (If `summary` scope present, with its special meaning, it should come first. Other scopes are reserved for multiple operations of a tool)
  * `functions` (multiple EDAM operation concepts describing 1 tool's operation or the `summary`)
  * `inputs`/`outputs` (multiple inputs/outputs of 1 operation)
  * `data` (multiple EDAM data concepts describing 1 operation's input/output)
  * `formats` (mutiple alternative formats of an operation's input/output)
* in the future, find correct EDAM IDs based on the version and the present primary term

Example:
```
edam:
  version: 1.25
  topics:
    - Sequence analysis
    - Phylogenetics
  - scope: summary
    functions:
      - Multiple sequence alignment
    inputs:
      - data:
        - Sequence
          formats:
            - FASTA format
            - EMBL format
    outputs:
      - data:
        - Sequence alignment (data)
          formats:
            - FASTA-aln
      - data:
        - Phylogenetic tree
          formats:
            - Phylip format
```

File: data/apollo/apollo.debian.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+identifiers:
+  - biotools: apollo
+License: Artistic
```
Comment (matuskalas) [inline data/apollo/apollo.debian.yaml]: Small letter for consistence, and sorted out of `identifiers`

**Comment by matuskalas** [comment]
@smoe and me also discussed that we can add more information from the UDD (Universal Debian Database):
 * Debian blend (such as Med, Ruby, _etc._, trivial)
 * Links (IDs) to _biii.eu_ and _conda-forge_ (@smoe works on it already)
 * List of binary packages for a source package? (nice-to-have but not trivial)
 *

**Comment by bgruening** [comment]
Awesome!!!! Hope you all have fun!!

File: data/apollo/apollo.debian.yaml (hunk 1 above)
Comment (matuskalas) [reply data/apollo/apollo.debian.yaml]: Solved by @smoe in 5713268. Note: The present licenses are only for non-official packages.

**Comment by bgruening** [comment]
I just love this! @smoe can we have the script that you used to generate this? We would like to run this script periodically on github actions.

**Comment by osallou** [comment]
Would be nice to put debian version AND upstream version i think

**Comment by smoe** [comment]
Quick comment on @matuskalas 's summary on
>       inputs: (and outputs alike)
>         - data:
>             - Sequence
>           formats:
>             - FASTA format
>             - EMBL format

We should give this some extra spin. There are different semantics between allowing one -data to feature just one "stream" (which may then be accepted in alternative formats) or to have multiple streams with a single data entry and then you are not completely confident what format belongs to which stream. I think I would prefer one -data entry for each stream. A threading implementation would then likely look much like 
```
       inputs: (and outputs alike)
         - data: Sequence
           formats:
             - FASTA format
             - EMBL format
         - data: Alignment
           formats:
             - FASTA format
             - Stockholm format
         - data: Structure
           formats:
             - PDB format
```
And should the -data structure possibly also have -name, and allow multiple entries for that?

**Comment by smoe** [comment]
> Would be nice to put debian version AND upstream version i think

I agree. At the Sprint we discussed the various deficiencies of Debian's package descriptions. Having too many of them being a meta-deficiency. If bio.tools gets it right, then for biologically motivated users Debian's pages, including our task pages, become obsolete. Working for it :o)

**Comment by smoe** [comment]
> I just love this! @smoe can we have the script that you used to generate this? We would like to run this script periodically on github actions.

Thank you for your kind words, @bgruening . All you need (?) is in https://github.com/bio-tools/biotoolsConnect/tree/master/DebianMed .  Since @matuskalas and I sat over this during the Sprint, a series of additional ideas came up to increase the coverage. I was saving those for the upcoming Hackathon. Send me some ePushes to assign them a negative nice level.

**Comment by matuskalas** [comment]
> We should give this some extra spin. There are different semantics between allowing one -data to feature just one "stream" (which may then be accepted in alternative formats) or to have multiple streams with a single data entry and then you are not completely confident what format belongs to which stream...

##### case 1: multiple I/O "streams"/"parameters"/files

Of course we need that. @smoe's last example right above has 3: a _Sequence_, an _Alignment_, and a _Structure_.

##### case 2: multiple EDAM Data concepts for one I/O "stream"/"parameter"/file

This is necessary in some cases, to describe what the given "stream"/"parameter"/file contains. Imagine a case where one of the output streams returns a list of genomic features (in GFF or similar), with additional transcription levels and RNA structure. Of course there isn't a single Data concept in EDAM covering all that, and annotating with only _Sequence features_ would be less than insufficient :-)

Something like (made-up):
```
      inputs:
        - data:
            - Genome annotation
            - RNA structure
            - Transcription levels
          formats:
            - GFF3
            - mirGFF
```

##### name

> And should the -data structure possibly also have -name, and allow multiple entries for that?

This would be a nice-to-have optional addition IMHO, useful in some cases. _E.g._:
```
      inputs:
        - name: query
          data:
            - Sequence
          formats:
            - FASTA format
        - name: target
          data:
            - Sequence
          formats:
            - FASTA format
```

What do you mean @smoe by _allow multiple entries for that_?

##### Debian

Not least but last: I guess that after we agree what we want here, we should implement that in Debian, right? Meaning updating the machinery of parsing it & storing in the UDD (Universal Debian Database), and reading from UDD (edam.sh) (?)

**Comment by joncison** [comment]
Following this thread with pleasure ...  thanks for the hard work.  I just chip in with one thing, because the semantics can be confusing, which is to highlight the difference between "function" and "operation" as defined / explained [here] (https://biotools.readthedocs.io/en/latest/curators_guide.html#toolfunctions) and in a more practical way [here](https://biotools.readthedocs.io/en/latest/community_specific_guidelines.html#operation).   Basically that a single "function" [== operation(s) with optional input(s) and output(s)] captures one modality of a tool. And some tools are multi-modal.   Not certain how this relates to streams, and forgive me if I'm stating the blindingly obvious!  Cheers!

**Pull request merged**